### DISCUSSION TITLE
Can we add generic session type to useSession and getSession?

### DISCUSSION THREAD

**Discussion by github0013** [nextauthjs/next-auth#2239] [Ideas]
Correct me if I am wrong, but because [we can add attributes to session object][1], do we want to pass generic type to `useSession` and `getSession`?

https://github.com/nextauthjs/next-auth/blob/fb04ab4e76cc16a877a730587e8bf5935633412f/types/client.d.ts#L26
https://github.com/nextauthjs/next-auth/blob/fb04ab4e76cc16a877a730587e8bf5935633412f/types/client.d.ts#L35

To something like this?

```ts
export function useSession<TSession extends Session>(): [TSession | null, boolean]

export function getSession<TSession extends Session>(options?: GetSessionOptions): Promise<TSession | null>
```

[1]: https://next-auth.js.org/configuration/callbacks#session-callback

**Comment by balazsorban44** [comment]
Check out this documentation for  augmenting your `Session` interface:

https://next-auth.js.org/getting-started/typescript
